### PR TITLE
Run smoke tests

### PR DESCRIPTION
**Pull request by patrickjahns**
Addresses #75

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -44,15 +44,59 @@ pipeline:
...
+
+  run-ui-smoke-test:
+    image: owncloudci/php
+    pull: true
+    environment:
+     - BROWSER=chrome
+     - SELENIUM_HOST=selenium
+     - SELENIUM_PORT=4444
+     - PLATFORM=Linux
+     - MAILHOG_HOST=docker
```
Comment (phil-davis) [inline .drone.yml]: I think this needs to be:
```
MAILHOG_HOST=mailhog
```

The name of the mailhog step becomes a hostname in the system pointing to the IP address given to that docker container.

Comment (patrickjahns) [reply .drone.yml]: That is however not routed from the outside - we are not using a drone service

We are building+starting the owncloud image in a "docker-in-docker" (dind) container. That is self-contained ( similar to a vm). 
So services from drone are not reachable from dind - but we can reach from the outside into dind with the published ports

Comment (patrickjahns) [reply .drone.yml]: with-in a user-define network 
https://github.com/owncloud-docker/server/pull/77/files#diff-3216dfff0ed3e301453e6799e8c367e2R50

and `docker` alias
https://github.com/owncloud-docker/server/pull/77/files#diff-3216dfff0ed3e301453e6799e8c367e2R83

it works now

**Comment by phil-davis** [comment]
I recently changed those ``editUser`` scenarios: https://github.com/owncloud/core/pull/33491/files
But, for example, fir ``editUser.feature:19`` it fails at a step that is before anything that I changed.
And other scenarios that change email and/or quota are working.

@individual-it what am I missing?

**Comment by patrickjahns** [comment]
@phil-davis - have there been any changes / additions that are only in stable10 but not in 10.0.10 ?

**Comment by phil-davis** [comment]
Found it, and got my memory back.
Issue https://github.com/owncloud/core/issues/33039
core ``master`` PR https://github.com/owncloud/core/pull/33040
core ``stable10`` backport https://github.com/owncloud/core/pull/33107

Previously to modify the ``displayname`` of a user, you had to send key ``display`` to the provisioning API endpoint. Now for consistency you can send key ``displayname``. The ``GET`` has always returned the attribute ``displayname``.

I guess this API enhancement happened after ``10.0.10`` release.

I will adjust the ``Smoketest`` tagged scenario so it uses ``display`` and have another scenario that uses ``displayname`` (because actually both are supposed to work nowadays, so we may as well test that)

**Comment by phil-davis** [comment]
core PR https://github.com/owncloud/core/pull/33523

File: .drone.yml [hunk 2]
```diff
@@ -136,3 +239,7 @@ pipeline:
 services:
   docker:
     image: docker:18.04-dind
```
Comment (tboerger) [inline .drone.yml]: Maybe add pull: true here as well

**Comment by patrickjahns** [comment]
Rebased - updated - failing scenarios for API tests related to webdavlocks

@phil-davis @individual-it

File: .drone.yml [hunk 3]
```diff
@@ -38,21 +38,126 @@ pipeline:
...
+
+  run-api-smoke-test:
+    image: owncloudci/php
+    pull: true
+    environment:
+      - TEST_SERVER_URL=http://docker:8000
+      - SKELETON_DIR=/mnt/data/apps/testing/data/apiSkeleton
+    commands:
+      - cd /drone/owncloud-smoketest/tests/acceptance
+      - bash run.sh --remote --tags "@smokeTest&&~@skip" --type api
```
Comment (individual-it) [inline .drone.yml]: should use `make test-acceptance-api` now

File: .drone.yml [hunk 4]
```diff
@@ -38,21 +38,126 @@ pipeline:
...
+      - SELENIUM_HOST=selenium
+      - SELENIUM_PORT=4444
+      - PLATFORM=Linux
+      - MAILHOG_HOST=email
+      - LOCAL_MAILHOG_HOST=docker
+      - TEST_SERVER_URL=http://docker:8000
+      - SKELETON_DIR=/mnt/data/apps/testing/data/webUISkeleton
+    commands:
+      - cd /drone/owncloud-smoketest/tests/acceptance
+      - bash run.sh --remote --tags "@smokeTest&&~@skip" --type webUI
```
Comment (individual-it) [inline .drone.yml]: should use `make test-acceptance-webui` now

File: .drone.yml (hunk 3 above)
Comment (patrickjahns) [reply .drone.yml]: how to pass the limitation on smokeTests there?

Comment (patrickjahns) [reply .drone.yml]: @individual-it 
makefile is not in the nightly tarball - thus need to rely on the command itself

**Comment by patrickjahns** [comment]
Now running into UI issues

```
    /drone/owncloud-smoketest/tests/acceptance/features/webUIAdminSettings/adminGeneralSettings.feature:56
    /drone/owncloud-smoketest/tests/acceptance/features/webUILogin/resetPassword.feature:29
    /drone/owncloud-smoketest/tests/acceptance/features/webUIManageUsersGroups/addUsers.feature:76
    /drone/owncloud-smoketest/tests/acceptance/features/webUIManageUsersGroups/addUsers.feature:77
```

**Comment by phil-davis** [comment]
adminGeneralSettings.feature:56 - that scenario has been there 5 months, tagged 4 months ago as smoketest. Not usre why it would fail now. Needs investigation.

resetPassword.feature:29
addUsers.feature:76
addUsers.feature:77
The password reset UI was recently (since 10.1) changed to have a retype password field to also fill in.
Now the question is, "what to do about tests that need to be adjusted because the workflow of the UI changes, but the test scenario is covering the same use case?"
In this case we could:
1) Tag the test scenario to skip on old versions - that means it will no longer run at all against those old versions
2) Keep a "copy" of the test methods that drive the old page, and a copy of the old test scenario, and tag it to only run on the old version/s of ownCloud
3) Decide that we should not expect to run the tests of 10.1 against 10.0, or the tests of upcoming 10.2 (current stable10) against 10.1 - e.g if you want to smoketest a 10.1 installation then use the set of acceptance test features/scenarios that were released along with 10.1
4) Or?

**Comment by patrickjahns** [comment]
> adminGeneralSettings.feature:56 - that scenario has been there 5 months, tagged 4 months ago as smoketest. Not usre why it would fail now. Needs investigation.

@phil-davis 
as said - in the docker container this is set via a Environment variable and a change from the webui doesn't have any effect. We need to skip this on the docker-containers

**Comment by phil-davis** [comment]
core https://github.com/owncloud/core/pull/34531 https://github.com/owncloud/core/pull/34532 should help. It should automagically skip the new retype-password scenarios. Specific `~@skipOnDockerContainerTesting` to skip the log level test. We can use that tag in future for any other tests that are not expected to work for the docker container.

**Comment by patrickjahns** [comment]
Used here:
https://github.com/owncloud-docker/server/pull/77/commits/2468bc06b46b005bebdfa7ca6ef8432df164bf7d

Just needs to have daily qa-tarball ready to include the new tags

File: .drone.yml [hunk 5]
```diff
@@ -157,7 +157,7 @@ pipeline:
       - SKELETON_DIR=/mnt/data/apps/testing/data/webUISkeleton
     commands:
       - cd /drone/owncloud-smoketest/tests/acceptance
-      - bash run.sh --remote --tags "@smokeTest&&~@skip" --type webUI
+      - bash run.sh --remote --tags "@smokeTest&&~@skip&&~@skipOnDockerContainerTestin" --type webUI
```
Comment (phil-davis) [inline .drone.yml]: s/skipOnDockerContainerTestin/skipOnDockerContainerTesting

Comment (patrickjahns) [reply .drone.yml]: Fixed - should all be green now 🤞

**Pull request merged**